modules/sfp_badpackets.py

```diff
@@ -0,0 +1,246 @@
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+        else:
+            self.results[eventData] = True
+
+        if eventName == 'NETBLOCK_OWNER':
+            if not self.opts['netblocklookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxnetblock']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxnetblock']))
+                    return None
+        
+        if eventName == 'NETBLOCK_MEMBER':
+            if not self.opts['subnetlookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxsubnet']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxsubnet']))
+                    return None
+                    
+        qrylist = list()
+        if eventName.startswith("NETBLOCK_"):
+            for ipaddr in IPNetwork(eventData):
+                qrylist.append(str(ipaddr))
+                self.results[str(ipaddr)] = True
+        else:
+            # If user has enabled affiliate checking
+            if eventName == "AFFILIATE_IPADDR" and not self.opts['checkaffiliates']:
+                return None
+            qrylist.append(eventData)
+        
+        for addr in qrylist:
+
+            nextPageHasData = True    
+            if self.checkForStop():
+                return None
+            
+            currentOffset = 0
+            while nextPageHasData:
+                data = self.queryIPAddress(addr, currentOffset)
+
+                if data is None:
+                    nextPageHasData = False
+                    break
+
+                count = data.get('count')
+                if count is None or int(count) == 0:
+                    nextPageHasData = False
+                    break
+
+                # Data is returned about the IP Address
+                evt = SpiderFootEvent("IP_ADDRESS", addr, self.__name__, event)
+                self.notifyListeners(evt)
+
+                results = data.get('results')
+                maliciousIPs = set()
+                for result in results:
+                    evt = SpiderFootEvent("RAW_RIR_DATA", str(result), self.__name__, event)
+                    self.notifyListeners(evt)
+                    self.reportExtraData(event, result)
+
+                    maliciousIP = result.get('source_ip_address')
+
+                    if maliciousIP:
+                        maliciousIPs.add(maliciousIP)
+
+                for maliciousIP in maliciousIPs:
+                    evt = SpiderFootEvent("MALICIOUS_IPADDR", str(maliciousIP), self.__name__, event)
```

Comment:
If you generate an `IP_ADDRESS` event because you received a `NETBLOCK_*` event, be sure to link that as the source event (last argument to this class initialisation), and NOT the event that `handleEvent()` received.